src/Tasks/Microsoft.NET.Build.Tasks/WriteAppConfig.cs

```diff
@@ -0,0 +1,153 @@
+
+        [Required]
+        public string TargetFramework { get; set; }
+
+        /// <summary>
+        /// Path to an intermediate file where we can write the input app.config plus the generated startup supportedRuntime
+        /// </summary>
+        [Required]
+        public ITaskItem OutputAppConfigFile { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            XDocument doc = LoadAppConfig(AppConfigFile);
+
+            AddSupportedRuntimeToAppconfigFile(doc, TargetFramework);
+
+            if (File.Exists(OutputAppConfigFile.ItemSpec))
+            {
+                File.Delete(OutputAppConfigFile.ItemSpec);
+            }
+
+            var fileStream = new FileStream(
+                OutputAppConfigFile.ItemSpec,
+                FileMode.Create,
+                FileAccess.Write,
+                FileShare.Read);
+            using (var stream = new StreamWriter(fileStream))
+            {
+                doc.Save(stream);
+            }
+        }
+
+        public static void AddSupportedRuntimeToAppconfigFile(XDocument doc, string targetframework)
+        {
+            XElement startupNode = doc.Root
+                                      .Nodes()
+                                      .OfType<XElement>()
+                                      .FirstOrDefault(e => e.Name.LocalName == "startup");
+
+            string runtimeVersion = string.Empty;
+            if (!HasExistingSupportedRuntime(startupNode))
+            {
+                if (TryGetSupportRuntimeNode(
+                    targetframework,
+                    runtimeVersion,
+                    out XElement supportedRuntime))
+                {
+                    if (startupNode == null)
+                    {
+                        startupNode = new XElement("startup");
+                        doc.Root.Add(startupNode);
+                    }
+
+                    startupNode.Add(supportedRuntime);
+                }
+            }
+        }
+
+        private static bool HasExistingSupportedRuntime(XElement startupNode)
+        {
+            return startupNode != null
+                  && startupNode.Nodes().OfType<XElement>().Any(e => e.Name.LocalName == "supportedRuntime");
+        }
+
+        //https://github.com/dotnet/docs/blob/master/docs/framework/configure-apps/file-schema/startup/supportedruntime-element.md
+        private static bool TryGetSupportRuntimeNode(string targetframework, string runtimeVersion, out XElement supportedRuntime)
+        {
+            supportedRuntime = null;
+            var targetFrameworkParsed = NuGetFramework.Parse(targetframework);
+            if (targetFrameworkParsed.Framework == ".NETFramework")
+            {
+                if (targetFrameworkParsed.Version.Major < 4)
+                {
+                    if (_targetFrameworkBelow40RuntimeVersionMap.TryGetValue(targetFrameworkParsed.GetShortFolderName(), out string value))
```

Comment:
Instead of using a map here, I would just use if statements, which check the range of values.  IE any version >= 2.0 but < 4.0 should use v2.0.50727 in the app.config.

Reply:
Fixed, check between version range